### DISCUSSION TITLE
Set position and size of input dialog box and button_label

### DISCUSSION THREAD

**Discussion by valentinavellani** [jspsych/jsPsych#1097] [Q&A] [answered]
Hi All,
I have a survey_text trial and I changed the position of the prompt.
I would like to change the position and size of the input dialog box and of the button_label as well.
I am using Gorilla so I would like to set these (size and position) inside the trial (as I did for the prompt).

```
var give_number = {
        type: 'survey-text',
        questions: [{prompt: '<span style="font-size:30px; position: absolute; top: 30%; with: 100%; left: 8%; text-align: center; ;">How much pollen do you think will be present on average in this garden? </span>', required: true, rows: 1, columns: 3}]
                     };
```

Thank you in advance,
Best
Valentina

**Comment by becky-gilbert** [comment, marked as answer]
Hi @valentinavellani! You can sort of change the text box size using the [survey-text question's `rows` and `columns` parameters](https://www.jspsych.org/plugins/jspsych-survey-text/#parameters). 

It's not possible to change other aspects of the text box and button (e.g. position, font size) directly through the plugin's parameters, so instead you can use CSS for this. You do need to know what the IDs are of the elements you want to change, which you can figure out by opening your developer tools and finding the element in the Elements tab.

One option is to add some custom CSS to your page by loading your own stylesheet or adding `<style>` tags to the `<head>` section. 

Another option is to use JavaScript to change the style of the elements on the page right after the trial has loaded. You can do inside of the [trial's `on_load` function](https://www.jspsych.org/overview/callbacks/#on_load). Here's an example:
```javascript
var give_number = {
        type: 'survey-text',
        questions: [{prompt: '<span style="font-size:30px; position: absolute; top: 30%; with: 100%; left: 8%; text-align: center; ;">How much pollen do you think will be present on average in this garden? </span>', required: true, rows: 1, columns: 3}],
        on_load: function() {
            var button = document.querySelector("#jspsych-survey-text-next");
            button.style.fontSize = "25px";
            button.style.padding = "25px";
            button.style.position = "absolute";
            button.style.top = "700px";
            button.style.left = "100px";
            var text_box = document.querySelector("#input-0");
            text_box.style.position = "absolute";
            text_box.style.left = "100px";
            text_box.style.fontSize = "30px";
            text_box.style.padding = "20px";
        }
    };
```

**Reply by valentinavellani** [reply]
Thanks Becky!
I think I will go with the last solution as I don't have too many trials in total.
I would have a question about the entry text plug in... When I click on the dialog to insert a number it shows all the things I have entered there in the past.  Since I need the participants to repeat the trial a number of times this could be an issue.
Do you know how to change this?

Thank you very much.

**Reply by becky-gilbert** [reply]
Yes, this is a known issue (#1064) that we're fixing in the next release 😃 In the meantime, you can either edit [this line of the survey-text plugin](https://github.com/jspsych/jsPsych/blob/9dfa38852de1454f59c3969b500717648f6a1c05/plugins/jspsych-survey-text.js#L103) so that the `<form>` tag includes `autocomplete="off"`, like this:
```javascript
html += '<form id="jspsych-survey-text-form" autocomplete="off">';
```
Or if you don't want to edit the plugin, you can do this in the trial's `on_load` function like this:
```javascript
var give_number = {
  type: 'survey-text',
  questions: [{prompt: '<span style="font-size:30px; position: absolute; top: 30%; with: 100%; left: 8%; text-align: center; ;">How much pollen do you think will be present on average in this garden? </span>', required: true, rows: 1, columns: 3}],
  on_load: function() {
    document.querySelector("#jspsych-survey-text-form").setAttribute('autocomplete','off');
  }
};
```

**Reply by valentinavellani** [reply]
Thank you very much!! :)
